ComponentKit/Core/CKComponentViewAttribute.h

```diff
@@ -56,14 +69,41 @@ struct CKComponentViewAttribute {
    |        |          | Updater is called if the attribute was previously applied and the value changes.              |
    |--------|----------|-----------------------------------------------------------------------------------------------|
    */
+  template<typename T, typename U, typename = typename std::enable_if<std::is_convertible<T, id>::value>::type>
   CKComponentViewAttribute(const std::string &ident,
-                           void (^app)(id view, id value),
-                           void (^unapp)(id view, id value) = nil,
-                           void (^upd)(id view, id oldValue, id newValue) = nil) :
+                           void (^app)(T view, U value),
+                           void (^unapp)(T view, U value),
+                           void (^upd)(T view, U oldValue, U newValue)) :
   identifier(ident),
-  applicator(app),
-  unapplicator(unapp),
-  updater(upd) {};
+  applicator([=](T view, U value) { app(view, value); }),
+  unapplicator(unapp ? ApplicatorFunc([=](T view, U value) { unapp(view, value); }) : ApplicatorFunc(nullptr)),
```

Comment:
I'm modifying the method signature to accept any callable type (function pointers, blocks, lambdas, functors).

Accepting an `std::function` directly here breaks brace initialization.

In addition, I still need to wrap this with a lambda as storing a block with signature `void (^)(UIView *, ...)` in an `std::function< void(id, ...) >` results in an `EXC_BAD` crash on the `std::function` destructor.